repository/repository.go

```diff
@@ -97,6 +97,10 @@ func (r *Repository) LoadBlob(t pack.BlobType, id backend.ID) ([]byte, error) {
 		return nil, err
 	}
 
+	if length > uint(cap(buf))+crypto.Extension {
+		return nil, errors.New("buf is too small")
```

Comment:
maybe include the number of bytes it needs to be able to hold, to make the error message more useful?